### PR TITLE
Avoid TypeErrror when action_buttons is None

### PR DESCRIPTION
**Pull request by HanlinMiao**
Closing #2107

### CODE REVIEW

File: nautobot/core/views/generic.py [hunk 1]
```diff
@@ -219,7 +219,9 @@ def validate_action_buttons(self, request):
         always_valid_actions = ("export",)
         valid_actions = []
         invalid_actions = []
-
+        # added check for whether the action_buttons exist because of issue #2107
+        if self.action_buttons is None:
+            self.actions_buttons = ("export",)
```
Comment (glennmatthews) [inline nautobot/core/views/generic.py:224]: 1. typo `action_buttons` versus `actions_buttons` - was this tested?
2. IMHO this function should fix up a `None` `action_buttons` to `[]`, not `("export",)`

File: nautobot/core/views/renderers.py [hunk 2]
```diff
@@ -85,8 +85,8 @@ def validate_action_buttons(self, view, request):
         valid_actions = []
         invalid_actions = []
         # added check for whether the action_buttons exist because of issue #2107
-        if not view.action_buttons:
-            view.actions_buttons = ("add", "import", "export")
+        if view.action_buttons is None:
+            view.actions_buttons = ("export",)
```
Comment (glennmatthews) [inline nautobot/core/views/renderers.py:89]: Same two comments here.

**Review by glennmatthews — CHANGES_REQUESTED** [review]

File: nautobot/core/views/generic.py (hunk 1 above)
Comment (HanlinMiao) [reply nautobot/core/views/generic.py:224]: 1. Sorry for the oversight, now this is tested and works.
2. Does this mean we dont consider "export" as an always valid action?

Comment (jathanism) [reply nautobot/core/views/generic.py]: > * typo `action_buttons` versus `actions_buttons` - was this tested?
> * IMHO this function should fix up a `None` `action_buttons` to `[]`, not `("export",)`

🤦

Comment (HanlinMiao) [reply nautobot/core/views/generic.py]: I know I cant believe I did that🫠

Comment (glennmatthews) [reply nautobot/core/views/generic.py]: `export` is always valid if it's specified, but not all views may want to support `export` - an empty list of action buttons is valid.

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**